beam/src/Rhyolite/Task/Beam.hs

```diff
@@ -0,0 +1,31 @@
+{-# Language DeriveAnyClass #-}
+{-# Language DeriveGeneric #-}
+{-# Language FlexibleInstances #-}
+{-# Language FunctionalDependencies #-}
+{-# Language KindSignatures #-}
+{-# Language RankNTypes #-}
+{-# Language TemplateHaskell #-}
+-- | Contains definitions for tasks, and tables that have tasks.
+module Rhyolite.Task.Beam where
+
+import Control.Lens.TH
+import Data.Text (Text)
+
+import Database.Beam
+
+-- TODO Update the documentation here
+-- | Data Family for tables that have tasks embedded in them.
+-- Specifically, this means tables that have the following 2 fields (columns)
+-- 1. A result field, containg an optional value
+-- 2. A checked out by field, containing the name of the worker that checked out this task
+-- For unclaimed tasks, both these fields are null (Nothing).
+-- When a task is checked out by a worker, the checkout out field is set to the worker's name. Result is still null.
+-- When a task is completed, the result field is filled with the result, and the checked out field is set to null.
+-- The type of result is dependent upon the task, hence the functional dependency between the table and the result type.
+data Task i o f = Task
+  { _taskPayload :: C f i
+  , _taskResult :: C f (Maybe o)
+  , _taskCheckedOutBy :: C f (Maybe Text)
```

Comment:
We might want to use a unique value from the database here, like the primary key of a worker registered in the database. One of the uses of this field is to be able to detect tasks that were checked out by workers that are no longer running (e.g., have crashed).

For instance, workers might register with the database, receive a primary key, use that key to check tasks out, and also use that key to report back to the database every _n_ minutes. If more then _n_ minutes (with some grace period) go by and we don't hear from the worker, then we may decide that it has died and needs to be replaced. In such a case, we'll look for tasks that worker checked out but didn't complete, and reset the statuses on those tasks.

Reply:
Done. The worker id is now generic.